nighres/brain/dots_segmentation.py

```diff
@@ -0,0 +1,704 @@
+       matter tracts using an overlapping fiber model." Proceedings of the 
+       International Workshop on Diffusion Modelling and Fiber Cup (2009)
+    """
+    
+    # Ignore runtime warnings that arise from trying to divide by 0/nan
+    # and all nan slices
+    np.seterr(divide = 'ignore', invalid = 'ignore')
+    
+    # Define the scalar constant c_I
+    c_I = 1/2    
+    
+      
+    # Create an array containing the directions between neighbors 
+    v_xy = np.zeros((3, 3, 3, 3))
+    for i in range(3):
+        for j in range(3):
+            for k in range(3):
+                if (i,j,k) == (1,1,1):
+                    v_xy[i,j,k,:] = np.nan
+                else:
+                    x = np.array([1,0,0])
+                    y = np.array([0,1,0])
+                    z = np.array([0,0,1])
+                    c = np.array([1,1,1])
+                    v_xy[i,j,k,:] = i*x + y*j + z*k - c
+                    v_xy[i,j,k,:] = v_xy[i,j,k,:] / \
+                                    np.linalg.norm(v_xy[i,j,k,:])
+    
+    
+    # Load tensor image
+    tensor_volume = load_volume(tensorimg).get_data()
+    
+    
+    # Load brain mask
+    brain_mask = load_volume(mask).get_data()
+    
+    
+    # Get dimensions of diffusion data
+    xs, ys, zs, _ = tensor_volume.shape
+    DWI_affine = load_volume(tensorimg).affine
+    
+    
+    # Calculate diffusion tensor eigenvalues and eigenvectors
+    tenfit = np.zeros((xs, ys, zs, 3, 3))
+    tenfit[:,:,:,0,0] = tensor_volume[:,:,:,0]
+    tenfit[:,:,:,1,1] = tensor_volume[:,:,:,1]
+    tenfit[:,:,:,2,2] = tensor_volume[:,:,:,2]
+    tenfit[:,:,:,0,1] = tensor_volume[:,:,:,3]
+    tenfit[:,:,:,1,0] = tensor_volume[:,:,:,3]
+    tenfit[:,:,:,0,2] = tensor_volume[:,:,:,4]
+    tenfit[:,:,:,2,0] = tensor_volume[:,:,:,4]
+    tenfit[:,:,:,1,2] = tensor_volume[:,:,:,5]
+    tenfit[:,:,:,2,1] = tensor_volume[:,:,:,5]
+    tenfit[np.isnan(tenfit)] = 0
+    evals, evecs = np.linalg.eig(tenfit)
+    idx = np.flip(np.argsort(evals, axis = 3), axis = 3)
+    evals = np.flip(np.sort(evals, axis = 3), axis = 3)
+    for i in range(xs):
+        for j in range(ys):
+            for k in range(zs):
+                evecs[i,j,k,:,:] = evecs[i,j,k,:,idx[i,j,k,:]]
+    evals[~brain_mask] = 0
+    evecs[~brain_mask] = 0
+
+    # Calculate FA
+    R = tenfit / np.trace(tenfit, axis1=3, axis2=4)[:,:,:,np.newaxis,np.newaxis]
+    FA = np.sqrt(0.5 * (3 - 1/(np.trace(np.matmul(R,R), axis1=3, axis2=4))))
+    FA[np.isnan(FA)] = 0    
+
+    
+    if wm_atlas == 1:
+                    
+        # Use smaller atlas
+        # Indices are
+        # 0 for isotropic regions
+        # 1 for unclassified white matter 
+        # 2-22 for individual tracts
+        # 22-73 for overlapping tracts
+        N_t = 23
+        N_o = 50        
+        print('N_t, N_o = '+str(N_t)+', '+str(N_o))
+        atlas_path = os.path.join(ATLAS_DIR, 'DOTS_atlas')
```

Comment:
Here and in other places in this file, the path is DOTS_atlas, while in the download module the defined path is dots_atlas. This way, the module cannot find the atlas. Either change in the downloader, or here